### ISSUE TITLE
Kubeconfig not used

### ISSUE THREAD

**Issue by maitredede** [docker/buildx#1697]
### Contributing guidelines

- [X] I've read the [contributing guidelines](https://github.com/docker/buildx/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Using buildx plugin v0.10.4, I am using a gitlab CI task to create a kubernetes builder. The kubernetes config file is at default place : `/root/.kube/config` in runner task. It is properly used (`kubectl get nodes` is ok).

When I want to create a builder `docker buildx create --bootstrap --driver=kubernetes [...]` I have an error `User "system:anonymous" cannot get resource "deployments" in API group "apps" in the namespace "gitlab-ci"`.

I also tried to set env var `KUBECONFIG`, but it does not change the behavior.

It may be related to https://github.com/docker/buildx/issues/1676

### Expected behaviour

The builder creation command `docker buildx create --bootstrap --driver=kubernetes [...]` should work.

I have downgraded to v0.9.1, it works :
- github.com/docker/buildx v0.9.1 ed00243a0ce2a0aee75311b06e32d33b44729689
- github.com/docker/buildx 2af40b7 2af40b75b7ec8818a8c30c1cf1b58e33a5288100

### Actual behaviour

Error `User "system:anonymous" cannot get resource "deployments" in API group "apps" in the namespace "gitlab-ci"`.
Tested with :
- github.com/docker/buildx v0.10.4 c513d34049e499c53468deac6c4267ee72948f02
- github.com/docker/buildx 948414e 948414e1b2c1f7707bb58b393ac45eccb1b372e8

### Buildx version

github.com/docker/buildx v0.10.4 c513d34049e499c53468deac6c4267ee72948f02

### Builders list

```markdown
Builder not yet created
```

### Configuration

```markdown
docker buildx create --name=mybuilder --platform=linux/amd64 --node=amd64
```
```


### Logs

_No response_

### Additional info

_No response_

**Comment by maitredede** [comment]
Based on #1676, https://github.com/docker/buildx/blob/4a73abfd645457ee8819855cb1a337ab23f833dc/builder/node.go#L80-L89 :

I have tried with a custom version of buildx with logging, and the error is pretty disapointing :
```
context "kubernetes:///ci-job-513721-nomarch?deployment=ci-job-513721-nomarch-amd64&kubeconfig=%2Froot%2F.kube%2Fconfig": context not found: open /root/.docker/contexts/meta/23abaa6b4bdd98bdf9056a5529007b5d3303031a61dff0664eca9aadc18d5e7e/meta.json: no such file or directory
```

**Comment by crazy-max** [comment]
@maitredede This should be fixed since 0.11.2 (https://github.com/docker/buildx/pull/1957)
